felix/bpf/proxy/proxy_test.go

```diff
@@ -536,6 +537,133 @@ var _ = Describe("BPF Proxy", func() {
 		},
 		)
 	})
+
+	Describe("with k8s client and Topology Aware Hints", func() {
+		Context("filter endpoints by corresponding zone", func() {
+			var (
+				p   proxy.Proxy
+				dp  *mockSyncer
+				k8s *fake.Clientset
+			)
+
+			testTopologySvc := &v1.Service{
+				TypeMeta: typeMetaV1("Service"),
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "customer",
+					Namespace: "yaobank",
+					Labels: map[string]string{
+						"app": "customer",
+					},
+					Annotations: map[string]string{
+						v1.AnnotationTopologyAwareHints: "Auto",
+					},
+				},
+				Spec: v1.ServiceSpec{
+					ClusterIP: "10.96.98.148",
+					Ports: []v1.ServicePort{
+						{
+							Name:       "http",
+							NodePort:   30180,
+							Port:       80,
+							Protocol:   v1.ProtocolTCP,
+							TargetPort: intstr.FromInt(80),
+						},
+					},
+					Type: v1.ServiceTypeNodePort,
+					Selector: map[string]string{
+						"app": "customer",
+					},
+				},
+			}
+
+			testTopologyEndpoint := &v1.Endpoints{
+				TypeMeta: typeMetaV1("Endpoints"),
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "customer",
+					Namespace: "yaobank",
+					Labels: map[string]string{
+						"app": "customer",
+					},
+				},
+				Subsets: []v1.EndpointSubset{
+					{
+						Addresses: []v1.EndpointAddress{
+							{
+								IP:       "192.168.152.3",
+								NodeName: strPtr("stevepro-bz-wyo0-worker2"),
+								TargetRef: &v1.ObjectReference{
+									Kind:      "Pod",
+									Name:      "customer-5dd6b8478-k9l5l",
+									Namespace: "yaobank",
+								},
+							},
+							{
+								IP:       "192.168.185.67",
+								NodeName: strPtr("stevepro-bz-wyo0-worker3"),
+								TargetRef: &v1.ObjectReference{
+									Kind:      "Pod",
+									Name:      "customer-5dd6b8478-dswzk",
+									Namespace: "yaobank",
+								},
+							},
+						},
+						Ports: []v1.EndpointPort{
+							{
+								Name:     "http",
+								Port:     80,
+								Protocol: v1.ProtocolTCP,
+							},
+						},
+					},
+				},
+			}
+
+			zones := []string{"us-west-2a", "us-west-2b"}
+			testTopologySvcEpsSlice := epsToSliceWithZones(testTopologyEndpoint, zones)
+
+			k8s = fake.NewSimpleClientset(testTopologySvc, testTopologySvcEpsSlice)
+
+			BeforeEach(func() {
+				By("creating proxy with fake client and mock syncer", func() {
+					var err error
+
+					syncStop = make(chan struct{})
+					dp = newMockSyncer(syncStop)
+
+					opts := []proxy.Option{proxy.WithImmediateSync()}
+
+					p, err = proxy.New(k8s, dp, "testnode", opts...)
+					Expect(err).NotTo(HaveOccurred())
+				})
+			})
+
+			AfterEach(func() {
+				By("stopping the proxy", func() {
+					close(syncStop)
+					p.Stop()
+				})
+			})
```

Comment:
isn't this already done?

Reply:
I was following the pattern - the other two Describe blocks have k8s | BeforeEach | AfterEach - thanks